Mathlib/MeasureTheory/Integral/SetIntegral.lean

```diff
@@ -403,6 +403,12 @@ theorem set_integral_eq_integral_of_forall_compl_eq_zero (h : ∀ x, x ∉ s →
   set_integral_eq_integral_of_ae_compl_eq_zero (eventually_of_forall h)
 #align measure_theory.set_integral_eq_integral_of_forall_compl_eq_zero MeasureTheory.set_integral_eq_integral_of_forall_compl_eq_zero
 
+lemma integral_eq_integral_restrict [MeasurableSpace E] [MeasurableSingletonClass E]
+    (hs : f =ᵐ[Measure.restrict μ sᶜ] 0) (f_mble : NullMeasurable f (μ.restrict sᶜ)) :
```

Comment:
You shouldn't need `f_mble` here, as it is a consequence of `hs`.
```suggestion
    (hs : f =ᵐ[restrict μ sᶜ] 0) :
```

Reply:
I failed to fix the proof without `(f_mble : NullMeasurable f μ)` or alternatively `(s_mble : MeasurableSet s)` (new lemma `integral_eq_integral_restrict'` below).

Could you clarify how you intended to bypass all measurability requirements?

Reply:
UPDATE: Sorry, I did not read my own NullMeasurability hypothesis correctly... This should be clear, I will fix!

Reply:
If a function `g` is null-measurable and a function `f` coincides ae with it, isn't `f` automatically null-measurable also? If not, discard my comment. If yes, apply this with `g = 0`.

Reply:
You are right, of course. Sorry!

I will also reverse the direction of the equality and try to have a naming more in line with the lemmas just above this, of the form `set_integral_eq_integral_of_...`.